### PR TITLE
Docs: Add archive for documentations older than 1.4.0

### PR DESCRIPTION
**Pull request by manuzhang**

### CODE REVIEW

**Comment by manuzhang** [comment]
cc @nastra @Fokko

File: site/docs/archive.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ - Unless required by applicable law or agreed to in writing, software
+ - distributed under the License is distributed on an "AS IS" BASIS,
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+
+Documentations of Iceberg versions older than 1.4.0 are no longer maintained. Here are the links to web archive.
+
+- [1.3.1](https://web.archive.org/web/20231210001939/https://iceberg.apache.org/docs/1.3.1/)
```
Comment (ajantha-bhat) [inline site/docs/archive.md:23]: I couldn't access this archive

image

Comment (ajantha-bhat) [reply site/docs/archive.md:23]: works now 👍

**Review by Fokko — APPROVED** [review]
Looks good, thanks @manuzhang for adding this, and @bitsondatadev for the review 🙌

**Pull request merged**